core/trino-main/src/test/java/io/trino/operator/scalar/TestSqlFunctions.java

```diff
@@ -0,0 +1,167 @@
+    private static final int[] intLefts = {9, 10, 11, -9, -10, -11, 0};
+    private static final int[] intRights = {3, -3};
+    private static final double[] doubleLefts = {9, 10, 11, -9, -10, -11, 9.1, 10.1, 11.1, -9.1, -10.1, -11.1};
+    private static final double[] doubleRights = {3, -3, 3.1, -3.1};
+    private static final double GREATEST_DOUBLE_LESS_THAN_HALF = 0x1.fffffffffffffp-2;
+
+    private QueryAssertions assertions;
+
+    @BeforeAll
+    public void init()
+    {
+        assertions = new QueryAssertions();
+    }
+
+    @AfterAll
+    public void teardown()
+    {
+        assertions.close();
+        assertions = null;
+    }
+
+    @Test
+    public void testFormatTrinoQuery()
+    {
+        // NULL input
+        assertThat(assertions.function("format_trino_query", "NULL"))
+                .isNull(VARCHAR);
+
+        // empty input
+        assertTrinoExceptionThrownBy(assertions.function("format_trino_query", "''")::evaluate)
+                .hasErrorCode(INVALID_FUNCTION_ARGUMENT)
+                .hasMessageStartingWith("Input cannot be parsed as a statement: []: line 1:2: mismatched input '<EOF>'. Expecting: ");
+
+        // input that is not SQL
+        assertTrinoExceptionThrownBy(assertions.function("format_trino_query", "'this is not sql'")::evaluate)
+                .hasErrorCode(INVALID_FUNCTION_ARGUMENT)
+                .hasMessageStartingWith("Input cannot be parsed as a statement: [this is not sql]: line 1:2: mismatched input 'this'. Expecting: ");
+
+        // trailing semicolon
+        assertTrinoExceptionThrownBy(assertions.function("format_trino_query", "'SELECT 1;'")::evaluate)
+                .hasErrorCode(INVALID_FUNCTION_ARGUMENT)
+                .hasMessageStartingWith("Input cannot be parsed as a statement: [SELECT 1;]: line 1:10: mismatched input ';'. Expecting: ");
+
+        // remove whitespace
+        assertFormatTrinoQuery("\n    \t  SELECT    1        \n\n\n", "SELECT 1");
+
+        // strip `-- ...` comments
+        assertFormatTrinoQuery("-- comment\nSELECT 1", "SELECT 1");
+
+        // strip `/* ... */` comments
+        assertFormatTrinoQuery("\n/* dbt\nlikes such\n comments*/ SELECT 1 ", "SELECT 1");
+
+        // SELECT FROM
+        assertFormatTrinoQuery(
+                "SELECT one_column FROM a_table",
+                """
+                SELECT one_column
+                FROM
+                  a_table""");
+
+        assertFormatTrinoQuery(
+                "SELECT col1, col2 FROM a_table",
+                """
+                SELECT
+                  col1
+                , col2
+                FROM
+                  a_table""");
+
+        // VALUES
+        assertFormatTrinoQuery(
+                " VALUES 1, 2, 3",
+                """
+                        VALUES\s
+                          1
+                        , 2
+                        , 3""");
+
+        // one aggregation
+        assertFormatTrinoQuery(
```

Comment:
This formatting is hard to read. We can make it better by doing the assertion inline. That also makes it easier to understand the tests:
```java
assertThat(formattedQuery("""
        SELECT ...
        """)
        .isEqualTo("""
        SELECT ...
        """)